File: public/dsa/searching/contains_element/lib/contains_element.py
from bisect import bisect_left
from typing import List


class Solution:
    def containsElement(self, arr: List[int], key: int) -> bool:
        index = bisect_left(arr, key)
        return index < len(arr) and arr[index] == key


# tests-start
arr = [1, 2, 3, 3, 3, 4, 4, 5]
key = 2
sol = Solution()
print(sol.containsElement(arr, key))
# tests-end

'''output
True
'''

File: public/dsa/searching/contains_element/lib/contains_element.cpp
#include <bits/stdc++.h>
using namespace std;

bool containsElement(vector<int> &arr, int key) {
    auto it = lower_bound(arr.begin(), arr.end(), key);
    return it != arr.end() && *it == key;
}

// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 3, 3, 4, 4, 5};
    int key = 2;

    cout << (containsElement(arr, key) ? "true" : "false") << endl;

    return 0;
}
// tests-end

/*output
true
*/
